lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -92,6 +92,20 @@ def authenticate(self, email, username, password, options):
                 log.debug('LDAP authenticate: email must be used to login, cannot be None')
                 return (failure_mode, '', '')
 
+        auto_create_roles = string_as_bool(options.get('auto-create-roles', False))
+        auto_create_groups = string_as_bool(options.get('auto-create-groups', False))
+        auto_create_roles_or_groups = auto_create_roles or auto_create_groups
+        auto_assign_roles_to_groups_only = string_as_bool(options.get('auto-assign-roles-to-groups-only', False))
+        if auto_assign_roles_to_groups_only and not (auto_create_roles and auto_create_groups):
+            raise ConfigurationError("If 'auto-assign-roles-to-groups-only' is True, auto-create-roles and "
+                                     "auto-create-groups have to be True as well.")
```

Comment:
These aren't LDAP-specific are they? If not, they should be handled in `AuthManager` I believe.